external-crates/move/crates/move-package/resolve_pkg_name_override/Move.toml

```diff
@@ -0,0 +1,22 @@
+#                                                             
+#             +----------+      2                               
+#    +------->|    B     |-------------+                        
+#    |        +----------+             |                        
+#    |                                 |                        
+#    |                                 v                        
+#  +----+     +----------+   3   +----------+                   
+#  |Root|---->| A-rename | ----->| C-rename |                   
+#  +----+     +----------+       +----------+                   
+#    |                                 ^                        
+#    |                                 |                        
+#    +---------------------------------+    
+#                  ov(1)
+#
+
+[package]
+name = "Root"
+
+[dependencies]
+A-rename = { local = "./deps_only/A-rename" }
+B = { local = "./deps_only/B" }
+C-rename = { local = "./deps_only/C-rename-v1", override = true }
```

Comment:
nit: this will compile locally, but typically you don't want to override in this way (to a lower version), as it won't publish.  Not an issue for this test, but may become an issue later if we become cleverer about detecting bad versioning arrangements locally.